### PR TITLE
Create background script

### PR DESCRIPTION
**Pull request by Bartleby2718**
### 1. Additional bundling through `webpack-cli` #43
- Probably the most important part of this pull request. 
- After the CRA build(`react-scripts build`) is over, `webpack` is used to bundle `node_modules` and `background.ts`. The downside is that it takes twice as long as the previous build. 
- It uses a separate tsconfig file--`tsconfig.other.json`--to transpile TS files because it requires that `noEmit` be false. Still, I minimized the difference by extending the existing `tsconfig.json`. #68
- In the process, I added `ts-loader` because it somehow works better than the existing `awesome-typescript-loader`.
### 2. Create folder upon installation #43 
- The code is basically the same as `saveClicked` or `loadClicked` because they all check the existence and the validity of `webmarkFolderId`.
### 3. Keyboard shortcuts #13
- Basically the opposite of #53
- Please test this on a Mac.
### 4. Refactorings #63 #66 #67
- Thanks to `utils.ts`, it is now very clear what happens in the popup (`index.tsx`).

### CODE REVIEW

File: public/manifest.json [hunk 1]
```diff
@@ -13,6 +13,28 @@
     "default_popup": "index.html",
     "default_icon": "images/default.png"
   },
+  "background": {
+    "scripts": [
+      "background.js"
```
Comment (jeongm-in) [inline public/manifest.json:18]: Is this supposed to be `background.js`? If so I believe it is the emitted result of `background.ts`.

File: tsconfig.other.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "noEmit": false,
```
Comment (jeongm-in) [inline tsconfig.other.json:4]: do you think we could extend `manifest.json` as well? I think it checks if file exists on runtime, and routed options page cannot be accessed then. 
But again I think it is impossible to use react-router anyways, as routing is done on the client side, not on the server side. Food for thought.

**Review by jeongm-in — COMMENTED** [review]
1. Additional bundling
- Need clarification on "works better." Does it mean performance improvement? Build time improvement?

2. On install folder creation
- confirmed folder is correctly created on install.
- new folder is created in whatever occasion it is gone.
3. Shortcut
- confirmed shortcuts are functional. 
- also tested on Ubuntu 18.04.2 LTS with latest Google Chrome stable 75.0.3770.100-1, although it doesn't mean much.
- will test on OSX environment later tonight.

File: public/manifest.json (hunk 1 above)
Comment (Bartleby2718) [reply public/manifest.json:18]: Good point. I haven't tried `.ts`, so I just did and got: `Could not load background script 'background.ts'.`
And Yes to the second sentence.

File: tsconfig.other.json (hunk 2 above)
Comment (Bartleby2718) [reply tsconfig.other.json:4]: There isn't anything regarding inheritance in https://developer.chrome.com/extensions/manifest, while there is in https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#configuration-inheritance-with-extends.

**Comment by Bartleby2718** [comment]
1. My bad. For some reason, I thought `awesome-typescript-loader` was included in the package but didn't work. It turns out it just wasn't installed. It works fine with `awesome-typescript-loader` once I `npm install --save awesome-typescript-loader`. Still, let's go with [`ts-loader`](https://github.com/TypeStrong/ts-loader) since it it shorter and more popular on GitHub than [`awesome-typescript-loader`](https://github.com/s-panferov/awesome-typescript-loader), although I don't know about the performance difference.

2, 3. Thank you for the testings.

**Comment by jeongm-in** [comment]
Shortcut works fine as expected in OSX. Approve merge into master
